### ISSUE TITLE
[BUG] - New version (3.10.1) of the library removes existing state machines.

### ISSUE THREAD

**Issue by danguitavinas** [serverless-operations/serverless-step-functions#531]
# This is a Bug Report

## Description

I deployed my application with some code changes, none of them were applied in the serverless.yml nor in the state machines. After the deployment, all of the state machines already created in previous deployments were deleted by cloudformation. Re-deploying after that does not seem to create them again.
After forcing the version of the library to: 3.10.0  the state machines were successfully re-deployed.

## Additional Data
* ***Serverless Framework Core Version you're using***: latest
* ***The Plugin Version you're using***: latest
* ***Operating System***: linux
* ***Stack Trace***: NONE
* ***Provider Error messages***: NONE

**Comment by horike37** [comment]
@danguitavinas 
Thank you for your report. We will take a look.

**Comment by lopburny** [comment]
Hello @danguitavinas , thank you for reporting this. Could you provide your serverless.yml sample for me to re-produce this issue?

**Comment by danguitavinas** [comment]
> Hello @danguitavinas , thank you for reporting this. Could you provide your serverless.yml sample for me to re-produce this issue?

Unfortunately I cannot share the code, it belongs to a private organization.

**Comment by maplesteve** [comment]
Several of our services fail to deploy because they reference the `id` of the state machine - which cannot be resolved by serverless.
The state machine config is in a separate file and it seems, that the current version of the plug-in just doesn't load the config. Validation is not performed, though it ist enabled in the config.
```yml
...
resources:
  - ${file(sls-cfg-resources/config1.yml)}
  - ${file(sls-cfg-resources/config2.yml)}
  - ${file(sls-cfg-resources/config3.yml)}

stepFunctions: ${file(sls-cfg-resources/timedOperationStateMachine.yml)}
```

**Comment by dbelikov-plana** [comment]
Same issues on my end with v3.10.1. GitHub workflows for state machines are failing with:
```
Serverless Error ----------------------------------------
 
The CloudFormation template is invalid: Template format error: Unresolved resource dependencies [***] in the Resources block of the template
```
Deployments with v3.10.0 were fine last Friday.

**Comment by yaronbenezra** [comment]
Please _do role back_ to this version 3.10.1 deploying with it erase all the stepfunction that was create before

**Comment by lopburny** [comment]
Thank you all, I just released a new version, `3.11.0` with reverting the change in yamlParser.js.

**Comment by danguitavinas** [comment]
> Thank you all, I just released a new version, `3.11.0` with reverting the change in yamlParser.js.

Thank you for the fast response!
